core/src/main/java/org/apache/iceberg/rest/responses/ConfigResponseParser.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.responses;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+public class ConfigResponseParser {
+
+  private static final String DEFAULTS = "defaults";
+  private static final String OVERRIDES = "overrides";
+
+  private ConfigResponseParser() {}
+
+  public static String toJson(ConfigResponse response) {
+    return toJson(response, false);
+  }
+
+  public static String toJson(ConfigResponse response, boolean pretty) {
+    return JsonUtil.generate(gen -> toJson(response, gen), pretty);
+  }
+
+  public static void toJson(ConfigResponse response, JsonGenerator gen) throws IOException {
+    Preconditions.checkArgument(null != response, "Invalid config response: null");
+
+    gen.writeStartObject();
+
+    JsonUtil.writeStringMap(DEFAULTS, response.defaults(), gen);
```

Comment:
existing tests in `TestConfigResponse` expect the maps to be always written (even if empty), so I omitted having a `isEmpty()` check

Reply:
Should we just change this to not send back the key in the response in the value is null (makes for a cleaner response? I don't really think that's a behavior change since in the end a client needs to check that the value is not null.

Reply:
NVM these two fields are spec'd out so it seems like servers should always send it even though it may be null 
https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L83